xdsl/dialects/arm/ops.py

```diff
@@ -196,3 +196,38 @@ def __init__(
 
     def assembly_line(self) -> str | None:
         return append_comment(f"{self.label.data}:", self.comment)
+
+
+@irdl_op_definition
+class CmpRegOp(ARMInstruction):
```

Comment:
Why CmpReg and not Cmp?

Reply:
the second operand could also be an immediate